### DISCUSSION TITLE
Alarm high_system_memory_usage is activated, System memory usage is higher than 70%. Why?

### DISCUSSION THREAD

**Discussion by josephthlee** [emqx/emqx#8402] [Q&A] [answered]
What triggers Alarm high_system_memory_usage is activated, System memory usage is higher than 70%?

Actually there was an issue opened in history but vague to me.
https://github.com/emqx/emqx/issues/4049

According to the issue, there are three memory values:

1. the total memory of the operating system
2. memory requested by Erlang VM
3. memory used by Erlang VM

My current environment: (emqx@172.31.27.227)
Total memory of OS: 2G.
emqx process VM 2G, resident 131MB
Memory Info on dashboard (used/total): 

![WechatIMG628](https://user-images.githubusercontent.com/30864375/177018978-1e5c12ba-bc20-476d-a938-f865f4368796.jpeg)

Sometimes there were high system memory usage alarms however I don't even have a clue for reason why.

1. What exactly triggers the 70% high memory usage? (Used/Total ?) 
    a. According to the snapshot above, 101.54 / 121.51 is about 0.83. (My alarm emits activated/deactivated couple times a day only) It doen't make sense to me. 
    b. It seems that the "total" in "Memory Info" is not fixed. Is it equivalent to the RSS of emqx process? What is the limitation of it? Or it just consumes the system memory until it OOM?
    c. What kind of command can check the current memory ratio?

2. How can I config the system more relative idle (less pressure) in terms of memory?

Thank you!

**Comment by zmstone** [comment, marked as answer]
Hi @josephthlee 
Which version is it?
We have recently had two fixes merged.
It should have been fixed in latest 4.3, 4.4 and 5.0

**Reply by josephthlee** [reply]
Thank @zmstone  for promptly response.

Current version is 4.3.5 on Ubuntu 20.04.
It was installed by APT from source https://repos.emqx.io/emqx-ce/deb/ubuntu/.
I think 4.3.5 is the latest stable update (from the APT resource)?
Which version should I upgrade to? Say 5.0.2?
Thank you!

**Reply by zmstone** [reply]
Hi 

4.3.16 or 4.4.5 both have the fix.
If you are open to a more non-compatible upgrade, you can try 5.0

apt source is a bit stale.
For now you can download the latest package from EMQX site:
E.g. 5.0.2
https://www.emqx.com/en/downloads-and-install?product=broker&version=5.0.2&os=Ubuntu20.04&oslabel=Ubuntu%2020.04

**Reply by josephthlee** [reply]
Thank @zmstone and I will try 5.0.2.
BTW, do you mean that it is a known bug in 4.3.5 and it should be fixed after the upgrade?

**Reply by zmstone** [reply]
Yes.
The alarm is a false alarm (most of the time), because the ram usage took reallocatable cache as “used”.
The fix was to count cache as available.
